core/payment/src/processor.rs

```diff
@@ -434,25 +446,33 @@ impl PaymentProcessor {
         // This is unconditional because at this point the invoice *has been paid*.
         // Whether the provider was correctly notified of this fact is another matter.
         counter!("payment.invoices.requestor.paid", 1);
-        let msg = SendPayment::new(payment, signature);
+
+        let msg = SendPayment::new(payment.clone(), signature);
+        let msg_with_bytes = SendSignedPayment::new(payment, signature_canonicalized);
 
         if payer_id != payee_id {
-            let mark_sent = ya_net::from(payer_id)
-                .to(payee_id)
-                .service(BUS_ID)
-                .call(msg)
-                .map(|res| match res {
-                    Ok(Ok(_)) => true,
-                    Err(err) => {
-                        log::error!("Error sending payment message to provider: {:?}", err);
-                        false
-                    }
-                    Ok(Err(err)) => {
-                        log::error!("Provider rejected payment: {:?}", err);
-                        true
-                    }
-                })
-                .await;
+            let send_result = Self::send_to_gsb(payer_id, payee_id, msg_with_bytes.clone()).await;
+
+            let mark_sent = if send_result.is_ok() {
+                payment_dao
+                    .add_signature(
+                        payment_id.clone(),
+                        msg_with_bytes.signature.clone(),
+                        msg_with_bytes.signed_bytes.clone(),
+                    )
+                    .await?;
+                true
+            } else if send_result.is_err_and(|err| err == SendToGsbError::NotSupported) {
+                // if sending SendPaymentWithBytes is not supported then try sending SendPayment
+                match Self::send_to_gsb(payer_id, payee_id, msg).await {
+                    Ok(_) => true,
+                    Err(SendToGsbError::Rejected) => true,
```

Comment:
I'm not sure what should we do.
What did previous code do?

Reply:
the boolean return value tells if we should mark as sent is db. This is consistent with previous behavior